compiler/damlc/lib/DA/Cli/Damlc.hs

```diff
@@ -792,15 +799,55 @@ execMigrate projectOpts opts0 inFile1_ inFile2_ mbDir = do
         errorOnLeft
             "Cannot decode daml-lf archive"
             (Archive.decodeArchive dalf)
-    getEntry fp dar =
-        maybe (fail $ "Package does not contain " <> fp) pure $
-        findEntryByPath fp dar
     getModule modName pkg =
         maybe
             (fail $ T.unpack $ "Can't find module" <> LF.moduleNameString modName)
             pure $
         NM.lookup modName $ LF.packageModules pkg
 
+-- | Get an entry from a dar or fail.
+getEntry :: FilePath -> Archive -> IO Entry
+getEntry fp dar =
+    maybe (fail $ "Package does not contain " <> fp) pure $
+    findEntryByPath fp dar
+
+-- | Merge two dars. The idea is that the second dar is a delta. Hence, we take the main in the
+-- manifest from the first.
+execMergeDars :: FilePath -> FilePath -> Maybe FilePath -> IO ()
+execMergeDars darFp1 darFp2 mbOutFp = do
+    let outFp = fromMaybe darFp1 mbOutFp
+    bytes1 <- B.readFile darFp1
+    bytes2 <- B.readFile darFp2
+    let dar1 = toArchive $ BSL.fromStrict bytes1
+    let dar2 = toArchive $ BSL.fromStrict bytes2
+    mf <- mergeManifests dar1 dar2
+    let merged =
+            Archive
+                (nubSortOn eRelativePath $ mf : zEntries dar1 ++ zEntries dar2)
```

Comment:
Might be a good idea to add a comment that you rely on the fact that `nubSortOn` keeps the first occurence.

Reply:
done.